lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -803,6 +771,90 @@ defmodule Thrift.Generator.StructBinaryProtocol do
     |> list_deserializer(name, file_group)
   end
 
+  def struct_serializer(%Union{fields: fields}, name, file_group) do
+    fields = Enum.reject(fields, &(&1.type == :void))
+
+    single_field_serializers = Enum.map(fields, fn
+      %Field{name: match_name} = match_field ->
+        nil_matchers = Enum.flat_map(fields, fn
```

Comment:
I'll leave it to your preference (and it looks like you'll rewrite it per below anyway). I was mostly just thinking about other ways to write that expression. 😊